internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -410,3 +356,45 @@ func (n *Networkd) SetReady() {
 
 	n.ready = true
 }
+
+func (n *Networkd) configureLinks(bonded bool) error {
+	errCh := make(chan error, len(n.Interfaces))
+	count := 0
+
+	for _, iface := range n.Interfaces {
+		if iface.Bonded != bonded {
+			continue
+		}
+
+		count++
+
+		go func(netif *nic.NetworkInterface) {
+			log.Printf("setting up %s", netif.Name)
+
+			errCh <- func() error {
+				// Ensure link exists
+				if err := netif.Create(); err != nil {
+					return fmt.Errorf("error creating nic: %v", err)
```

Comment:
I guess we need to add NIC name to the error message to make it easy to match error & config

Reply:
Good point